airflow/gcp/hooks/base.py

```diff
@@ -323,8 +341,63 @@ def provide_gcp_credential_file_as_context(self):
                     pass
                 yield conf_file
             finally:
-                if current_env_state is None:
-                    if CREDENTIALS in os.environ:
-                        del os.environ[CREDENTIALS]
-                else:
-                    os.environ[CREDENTIALS] = current_env_state
+                self.restore_env_variable(current_env_state)
+
+    @staticmethod
+    def build_gcp_conn(
+        key_file_path: Optional[str] = None,
+        scopes: Optional[Sequence[str]] = None,
+        project_id: Optional[str] = None,
+        gcp_conn_id: str = 'google_cloud_default'
+    ) -> str:
+        """
+        Builds a variable that can be used as `AIRFLOW_CONN_GOOGLE_CLOUD_DEFAULT` with provided service key,
+        scopes and project id.
+        """
+        conn = "google-cloud-platform://?"
+        extras = "extra__{}".format(gcp_conn_id)
```

Comment:
It's will probable doesn't work. This contents is related to UI.

Reply:
What do you mean by not working?

Reply:
The program will not set the appropriate value. Look at the documentation.
https://airflow.apache.org/howto/connection/gcp.html
and in source code:
https://github.com/apache/airflow/blob/master/airflow/www/static/js/connection_form.js#L29

Reply:
I think I still miss something. This works at least when I run system tests.

Reply:
This only works with the default connection ID. https://github.com/apache/airflow/blob/master/airflow/gcp/hooks/base.py#L163-L174